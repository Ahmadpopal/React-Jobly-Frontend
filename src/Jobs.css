.Jobs-div-title {
  margin: 10px;
}

.Jobs-title {
  margin: 10px;
  text-align: center;
}
