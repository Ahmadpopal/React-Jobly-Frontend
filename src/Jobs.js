import React, { useState, useEffect } from 'react';
import { Container, Row, Col } from 'reactstrap';

import JoblyApi from './Jobly'
import JobCardList from './JobCardList'
import SearchForm from './helper/SearchForm'
import './Jobs.css'


// Jobs Component: 
//   - Show Jobs list from Jobly Api 
//   - show Search Form where Users can search jobs by passing job title  

// Renders 
//   - SearchForm Component 
//   - JobCard Component

const Jobs = () => {

const [jobs, setJobs] = useState(null);

useEffect(() =>{
    getAllJobs()
}, [])

async function getAllJobs(search){
    let jobs = await JoblyApi.getJobs(search)
    setJobs(jobs)
}

if(!jobs){
    return <h2> Job List Loading.... </h2>
}

return (
    <Container className="Jobs">
        <Row>
            <Col className="Jobs-div-title">
                <h3 className="Jobs-title"> Jobs List <hr/></h3>
            </Col>
        </Row>
        <Row>
          <Col>
            <SearchForm searchFor={getAllJobs} />
          </Col>
        </Row>
            <JobCardList jobs={jobs}/>
    </Container>
    )
}


export default Jobs