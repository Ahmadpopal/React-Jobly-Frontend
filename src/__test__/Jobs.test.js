import React from 'react';
import { render } from '@testing-library/react';
import Jobs from '../Jobs';
import { MemoryRouter, Route } from 'react-router-dom';

describe('JOBS TESTS ', () => {
	it('Jobs Component should render without crashing', () => {
		render(
			<MemoryRouter initialEntries={[ '/jobs/' ]}>
				<Route exact path="/jobs/">
					<Jobs />
				</Route>
			</MemoryRouter>
		);
	});

	it('Jobs matches snapshot', () => {
		const { asFragment } = render(
			<MemoryRouter initialEntries={[ '/jobs/' ]}>
				<Route exact path="/jobs/">
					<Jobs />
				</Route>
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});