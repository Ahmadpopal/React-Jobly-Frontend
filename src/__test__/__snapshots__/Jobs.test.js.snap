// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JOBS TESTS  Jobs matches snapshot 1`] = `
<DocumentFragment>
  <h2>
     Job List Loading.... 
  </h2>
</DocumentFragment>
`;
